flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -145,12 +157,43 @@ public void shutdown(Time timeout) {
 		}
 	}
 
-	public <M extends MessageHeaders<R, P, U>, U extends MessageParameters, R extends RequestBody, P extends ResponseBody> CompletableFuture<P> sendRequest(String targetAddress, int targetPort, M messageHeaders, U messageParameters, R request) throws IOException {
+	public <M extends MessageHeaders<R, P, U>, U extends MessageParameters, R extends RequestBody, P extends ResponseBody> CompletableFuture<P> sendRequest(
+		String targetAddress,
+		int targetPort,
+		M messageHeaders,
+		U messageParameters,
+		R request) throws IOException {
+		return internalSendRequest(targetAddress, targetPort, messageHeaders, messageParameters, request, false, new DefaultProcessor());
```

Comment:
I think here we should call `sendRequest(targetAddress, targetPort, messageHeaders, messageParameters, request, Collections.emptyList())`